### PR TITLE
importer: fix overflow and add support for TypeNewDecimal

### PR DESCRIPTION
**Pull request by fipped**
@hanfei1991 @coocood PTAL

### CODE REVIEW

**Comment by hanfei1991** [comment]
LGTM

File: cmd/importer/db.go [hunk 1]
```diff
@@ -209,6 +209,16 @@ func genColumnData(table *table, column *column) (string, error) {
 
 		data = append(data, '\'')
 		return string(data), nil
+	case mysql.TypeNewDecimal:
+		var data float64
+		var limit = int64(math.Pow10(tp.Flen))
+		var scale = math.Pow10(-tp.Decimal)
+		if isUnique {
```
Comment (lamxTyler) [inline cmd/importer/db.go]: Should we also consider `unsigned`?

File: cmd/importer/db.go [hunk 2]
```diff
@@ -209,6 +209,16 @@ func genColumnData(table *table, column *column) (string, error) {
...
+			data = float64(uniqInt64Value(column, -limit, limit)) * scale
+		} else {
+			data = float64(randInt64Value(column, -limit, limit)) * scale
+		}
+		return strconv.FormatFloat(data, 'f', -1, 64), nil
```
Comment (coocood) [inline cmd/importer/db.go]: Sometimes, we need to use NewDecimal in index, convert it to float makes equal decimal not match.

**Comment by shenli** [comment]
@coocood @lamxTyler PTAL

File: cmd/importer/db.go [hunk 3]
```diff
@@ -209,6 +210,32 @@ func genColumnData(table *table, column *column) (string, error) {
 
 		data = append(data, '\'')
 		return string(data), nil
+	case mysql.TypeNewDecimal:
+		var data string
+		var limit = int64(math.Pow10(tp.Flen))
```
Comment (coocood) [inline cmd/importer/db.go]: When `math.Pow10` is larger than max int64, cast it to int64 will get a negative value.

File: cmd/importer/db.go [hunk 4]
```diff
@@ -209,6 +210,32 @@ func genColumnData(table *table, column *column) (string, error) {
...
+		if isUnique {
```
Comment (coocood) [inline cmd/importer/db.go]: We can extract a function like
```go
func intToDecimalString(intValue int64, decimal int) string 
```

And write unit test for it.

File: cmd/importer/db.go [hunk 5]
```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/juju/errors"
 	"github.com/pingcap/tidb/mysql"
 	log "github.com/sirupsen/logrus"
+	"strings"
```
Comment (lamxTyler) [inline cmd/importer/db.go]: Put it upward.

File: cmd/importer/db_test.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (lamxTyler) [inline cmd/importer/db_test.go]: 2018.

File: cmd/importer/db.go [hunk 7]
```diff
@@ -209,6 +228,22 @@ func genColumnData(table *table, column *column) (string, error) {
...
+		if limit < 0 {
+			limit = math.MaxInt64
+		}
+		if isUnique {
+			intVal = uniqInt64Value(column, 0, limit-1)
+		} else {
+			if isUnsigned {
+				intVal = randInt64Value(column, 0, limit-1)
+			} else {
+				intVal = randInt64Value(column, -limit+1, limit-1)
```
Comment (lamxTyler) [inline cmd/importer/db.go]: If the `limit` is `MaxInt64`, will this cause panic?

**Comment by coocood** [comment]
LGTM

**Pull request merged**